Crash while trying to open a downloaded video

### Checklist

- [x] I am able to reproduce the bug with the latest version given here: [CLICK THIS LINK](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [x] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [x] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [x] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [x] This issue contains only one bug.
- [x] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.28.0

### Steps to reproduce the bug

1. Go to 'Downloads'
2. Press on 'the Downloaded Video'

### Expected behavior

It should directly open the video in the default selected video player app or ask in which video player app should the video be played in.

### Actual behavior

Whenever I press on the Downloaded video to open & play it , Newpipe app crashes.

### Screenshots/Screen recordings

_No response_

### Logs

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ CA
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ none
* __Timestamp:__ 2025-12-14T08:51:50.949+05:30
* __Package:__ org.schabi.newpipe
* __Service:__ none
* __Version:__ 0.28.0
* __OS:__ Linux Android 12 - 31
<details><summary><b>Crash log </b></summary><p>

```
java.lang.SecurityException: UID 10235 does not have permission to content://com.android.externalstorage.documents/document/primary%3ADownload%2FTelegram%2FIncrease%20Your%20Internet%20Speed%20!%20-%20Jio%2C%20Airtel%2C%20Vi%20%26%20BSNL%20(100%25%20Working).mp4 [user 0] [prefix]; you could obtain access using ACTION_OPEN_DOCUMENT or related APIs
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2440)
	at android.os.Parcel.createException(Parcel.java:2424)
	at android.os.Parcel.readException(Parcel.java:2407)
	at android.os.Parcel.readException(Parcel.java:2349)
	at android.app.IActivityTaskManager$Stub$Proxy.startActivity(IActivityTaskManager.java:2847)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1793)
	at android.app.Activity.startActivityForResult(Activity.java:5634)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:780)
	at android.app.Activity.startActivityForResult(Activity.java:5584)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:761)
	at android.app.Activity.startActivity(Activity.java:6002)
	at android.app.Activity.startActivity(Activity.java:5955)
	at org.schabi.newpipe.util.external_communication.ShareUtils.tryOpenIntentInApp(ShareUtils.java:145)
	at org.schabi.newpipe.util.external_communication.ShareUtils.openIntentInApp(ShareUtils.java:163)
	at us.shandian.giga.ui.adapter.MissionAdapter.viewWithFileProvider(MissionAdapter.java:356)
	at us.shandian.giga.ui.adapter.MissionAdapter.-$$Nest$mviewWithFileProvider(MissionAdapter.java:0)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.lambda$new$1(MissionAdapter.java:899)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.$r8$lambda$lUyMuzEHgx7VO8u56WYBu3QggdM(MissionAdapter.java:0)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem$$ExternalSyntheticLambda1.onClick(R8$$SyntheticClass:0)
	at android.view.View.performClick(View.java:7512)
	at android.view.View.performClickInternal(View.java:7489)
	at android.view.View.access$3700(View.java:857)
	at android.view.View$PerformClick.run(View.java:29031)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:233)
	at android.os.Looper.loop(Looper.java:334)
	at android.app.ActivityThread.main(ActivityThread.java:8399)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:582)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1068)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionUnlocked(UriGrantsManagerService.java:1320)
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:648)
	at com.android.server.uri.UriGrantsManagerService.access$1000(UriGrantsManagerService.java:114)
	at com.android.server.uri.UriGrantsManagerService$LocalService.checkGrantUriPermissionFromIntent(UriGrantsManagerService.java:1541)
	at com.android.server.wm.ActivityStarter$Request.resolveActivity(ActivityStarter.java:640)


```
</details>
<hr>


### Affected Android/Custom ROM version

Android 12 / Funtouch OS 12 Global 

### Affected device model

Vivo Y51 (V2030)

### Additional information

I did allow all the applicable permission(s) as I was told in the previous bug report by the expert but it's still the same , whenever I try to play a downloaded video the Newpipe App crashes. Please help , I'm a big fan of this  app.

--- a/app/src/main/java/us/shandian/giga/ui/adapter/MissionAdapter.java
+++ b/app/src/main/java/us/shandian/giga/ui/adapter/MissionAdapter.java
@@ -2,6 +2,7 @@ package us.shandian.giga.ui.adapter;
 
 import static android.content.Intent.FLAG_GRANT_PREFIX_URI_PERMISSION;
 import static android.content.Intent.FLAG_GRANT_READ_URI_PERMISSION;
+import static android.content.Intent.createChooser;
 import static us.shandian.giga.get.DownloadMission.ERROR_CONNECT_HOST;
 import static us.shandian.giga.get.DownloadMission.ERROR_FILE_CREATION;
 import static us.shandian.giga.get.DownloadMission.ERROR_HTTP_NO_CONTENT;
@@ -349,11 +350,15 @@ public class MissionAdapter extends Adapter<ViewHolder> implements Handler.Callb
         if (BuildConfig.DEBUG)
             Log.v(TAG, "Mime: " + mimeType + " package: " + BuildConfig.APPLICATION_ID + ".provider");
 
-        Intent intent = new Intent(Intent.ACTION_VIEW);
-        intent.setDataAndType(resolveShareableUri(mission), mimeType);
-        intent.addFlags(FLAG_GRANT_READ_URI_PERMISSION);
-        intent.addFlags(FLAG_GRANT_PREFIX_URI_PERMISSION);
-        ShareUtils.openIntentInApp(mContext, intent);
+        Intent viewIntent = new Intent(Intent.ACTION_VIEW);
+        viewIntent.setDataAndType(resolveShareableUri(mission), mimeType);
+        viewIntent.addFlags(FLAG_GRANT_READ_URI_PERMISSION);
+        viewIntent.addFlags(FLAG_GRANT_PREFIX_URI_PERMISSION);
+
+        Intent chooserIntent = createChooser(viewIntent, null);
+        chooserIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | FLAG_GRANT_READ_URI_PERMISSION);
+
+        ShareUtils.openIntentInApp(mContext, chooserIntent);
     }
 
     private void shareFile(Mission mission) {
@@ -364,8 +369,7 @@ public class MissionAdapter extends Adapter<ViewHolder> implements Handler.Callb
         shareIntent.putExtra(Intent.EXTRA_STREAM, resolveShareableUri(mission));
         shareIntent.addFlags(FLAG_GRANT_READ_URI_PERMISSION);
 
-        final Intent intent = new Intent(Intent.ACTION_CHOOSER);
-        intent.putExtra(Intent.EXTRA_INTENT, shareIntent);
+        final Intent intent = createChooser(shareIntent, null);
         // unneeded to set a title to the chooser on Android P and higher because the system
         // ignores this title on these versions
         if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.O_MR1) {

